numpy/core/src/multiarray/datetime.c

```diff
@@ -2457,8 +2457,14 @@ convert_pyobject_to_datetime(PyArray_DatetimeMetaData *meta, PyObject *obj,
         Py_DECREF(bytes);
         return 0;
     }
+
     /* Do no conversion on raw integers */
-    else if (PyInt_Check(obj) || PyLong_Check(obj)) {
+    PyObject *type, *value, *traceback;
+    PyErr_Fetch(&type, &value, &traceback);
+    npy_intp intp = PyArray_PyIntAsIntp(obj);
+    PyErr_Restore(type, value, traceback);
```

Comment:
You shouldn't need to do this - you know there's no error getting here, so the `PyErr_Fetch` is not going to get anything.

At that point,  the `PyErr_Restore` is the same as `PyErr_Clear`.

You'll want to clear it only after you do the `error_converting` check.